### PR TITLE
LS-54470: delete streams from terraform state only

### PR DESCRIPTION
**Pull request by ltyson**
What does this PR do and why?
-------------------------------
Changes the behavior of the Lightstep terraform provider to reduce friction when dealing with streams that are defined in terraform and implicitly used in other resources (most notably: unified alerts)

See [this thread](https://lightstep.slack.com/archives/C8QMQNV46/p1697036359432489) for additional detail about the use-case. 

How did you test this?
-------------------------------
Updated the acceptance tests to exercise this scenario (and fixed a few test bugs along the way)

R: @lambcode @mattcarmody @DanTulovsky @matthagenbuch

### CODE REVIEW

File: lightstep/resource_stream.go [hunk 1]
```diff
@@ -144,7 +146,16 @@ func resourceStreamDelete(ctx context.Context, d *schema.ResourceData, m interfa
 
 	c := m.(*client.Client)
 	if err := c.DeleteStream(ctx, d.Get("project_name").(string), d.Id()); err != nil {
-		return diag.FromErr(fmt.Errorf("failed to delete stream: %v", err))
+		if apiClientError, ok := err.(client.APIResponseCarrier); ok && apiClientError.GetStatusCode() == http.StatusConflict {
+			// Lightstep didn't delete the stream itself because there are other resources
+			// (usually alerts) that depend on it. However, that relationship is often implicit,
+			// and thus opaque to the terraform user, so we do our best to satisfy the user's
+			// intent by acting as though the delete operation succeed. This will cause the stream
```
Comment (mattcarmody) [inline lightstep/resource_stream.go:153]: nit: succeed to succeeded?

File: lightstep/resource_stream_test.go [hunk 2]
```diff
@@ -114,6 +116,149 @@ resource "lightstep_stream" "import-stream"{
 	})
 }
 
+func TestAccDeleteStreamWithDependentResources(t *testing.T) {
+	streamConfig := `
+	resource "lightstep_stream" "web_errors" {
+	  project_name = "` + testProject + `"
+      stream_name = "web errors"
```
Comment (mattcarmody) [inline lightstep/resource_stream_test.go:123]: nit: inconsistent indent

File: lightstep/resource_stream_test.go [hunk 3]
```diff
@@ -114,6 +116,149 @@ resource "lightstep_stream" "import-stream"{
 	})
 }
 
+func TestAccDeleteStreamWithDependentResources(t *testing.T) {
+	streamConfig := `
```
Comment (mattcarmody) [inline lightstep/resource_stream_test.go:120]: nit/personal preference: if there's a streamConfig2 I often appreciate when the first is numbered

File: lightstep/resource_stream_test.go [hunk 4]
```diff
@@ -114,6 +116,149 @@ resource "lightstep_stream" "import-stream"{
...
+	  thresholds {
+		critical  = 10
+		warning = 5
+	  }
+  }
+
+  query {
+    hidden              = false
+    query_name          = "a"
+    display = "line"
```
Comment (mattcarmody) [inline lightstep/resource_stream_test.go:157]: nit: padding before =

**Review by mattcarmody — APPROVED** [review]
🎉

File: lightstep/resource_stream.go (hunk 1 above)
Comment (ltyson) [reply lightstep/resource_stream.go]: ✅

File: lightstep/resource_stream_test.go (hunk 2 above)
Comment (ltyson) [reply lightstep/resource_stream_test.go]: fixed

File: lightstep/resource_stream_test.go (hunk 3 above)
Comment (ltyson) [reply lightstep/resource_stream_test.go]: done!

File: lightstep/resource_stream_test.go (hunk 4 above)
Comment (ltyson) [reply lightstep/resource_stream_test.go]: fixed!

**Pull request merged**